src/App.js

```diff
@@ -1,26 +1,45 @@
-import React from 'react';
-import logo from './logo.svg';
+import React, { Component } from 'react'
 import './App.css';
+import Header from './Header.js';
+import ImageList from './ImageList.js';
+import ImageItem from './ImageItem.js';
+import images from './data.js';
 
-function App() {
-  return (
-    <div className="App">
-      <header className="App-header">
-        <img src={logo} className="App-logo" alt="logo" />
-        <p>
-          Edit <code>src/App.js</code> and save to reload.
-        </p>
-        <a
-          className="App-link"
-          href="https://reactjs.org"
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          Learn React
-        </a>
-      </header>
-    </div>
-  );
+export default class App extends Component {
+
+  //intialize state to null 
+  state = { selected: null }; 
+
+  render() {
+    return (
```

Comment:
you haven't learned about component state scope yet. for now you need to know that your state-setting functions need to be in the parent component that controls the state. then you can pass the whole handleChange function down as a prop to your image list component. if that function is moved here and it is passed down properly, your app works great!